tests/js/common/shell/shell-pregel-components.js

```diff
@@ -200,29 +200,20 @@ function componentsTestSuite() {
 
         testWCC: function () {
             var pid = pregel.start("wcc", graphName, {resultField: "result", store: true});
-            var i = 10000;
-            do {
-                internal.sleep(0.2);
-                let stats = pregel.status(pid);
-                if (stats.state !== "loading" && stats.state !== "running" && stats.state !== "storing") {
-                    assertEqual(stats.vertexCount, numComponents * n, stats);
-                    assertEqual(stats.edgeCount, numComponents * (m + n), stats);
-
-                    let c = db[vColl].all();
-                    let mySet = new Set();
-                    while (c.hasNext()) {
-                        let doc = c.next();
-                        assertTrue(doc.result !== undefined, doc);
-                        mySet.add(doc.result);
-                    }
-                    assertEqual(mySet.size, numComponents);
-
-                    break;
-                }
-            } while (i-- >= 0);
-            if (i === 0) {
-                assertTrue(false, "timeout in WCC execution");
+            const stats = pregelTestHelpers.waitUntilRunFinishedSuccessfully(pid);
+
+            assertEqual(stats.vertexCount, numComponents * n, stats);
+            assertEqual(stats.edgeCount, numComponents * (m + n), stats);
+
+            let c = db[vColl].all();
+            let mySet = new Set();
+            while (c.hasNext()) {
+                let doc = c.next();
+                assertTrue(doc.result !== undefined, doc);
+                mySet.add(doc.result);
```

Comment:
I see this code repeated a few times; maybe factor it into separate function?

Reply:
Not sure if I found now all occurences: I found 4 and replaced them with a function.